src/components/tutorials/TutorialList/index.tsx

```diff
@@ -0,0 +1,294 @@
+          </StartButton>
+        </Grid>
+      </Grid>
+    </Card>
+  )
+}
+
+function searchFilter(term: string, tutorial?: Tutorial): boolean {
+  return tutorial
+    ? tutorial?.meta.title.toLowerCase().includes(term.toLowerCase()) ||
+        tutorial?.meta.description.toLowerCase().includes(term.toLowerCase())
+    : false
+}
+
+function topicFilter(topic: TopicDropdown, tutorial?: Tutorial): boolean {
+  if (!tutorial) return false
+  if (topic === 'All topics') return true
+  return tutorial ? tutorial?.meta.label === topic : false
+}
+
+type TopicDropdown = keyof typeof TopicType.Values | 'All topics'
+const TopicDropdownValues: TopicDropdown[] = [
+  'All topics',
+  ...Object.values(TopicType.Values),
+]
+
+const TutorialList: FC = () => {
+  const theme = useTheme()
+  const isMobile = useMediaQuery(theme.breakpoints.down('md'))
+
+  const [search, setSearch] = useState('')
+  const [topic, setTopic] = useState<TopicDropdown>('All topics')
+  const [parsedTutorials, setParsedTutorials] = useState<Tutorial[]>([])
+  const [tutorials, _] = useState<Meta[]>(() => getMetaData())
+
+  useEffect(() => {
+    setParsedTutorials(
+      Object.values(tutorials).map((metaJson) => {
+        const meta = Meta.parse(metaJson)
+        const steps = getSteps(meta.id)
+        const tutorial = { meta, steps }
+        const parsedTutorials = Tutorial.parse(tutorial)
+
+        // Ensure no duplicate positions
+        const duplicates = new Set<number>()
+        for (const step of parsedTutorials.steps) {
+          if (duplicates.has(step.position)) {
+            throw new Error(
+              `Duplicate step position ${step.position} in tutorial "${parsedTutorials.meta.id}"` +
+                `\nCheck steps: \n${parsedTutorials.steps
+                  .filter((s) => s.position === step.position)
+                  .map((s) => s.path)
+                  .join('\n')}\n`
+            )
+          }
+          duplicates.add(step.position)
+        }
+
+        return parsedTutorials
+      })
+    )
+  }, [tutorials])
+
+  return isMobile ? (
+    <MobileTutorialList
+      search={search}
+      setSearch={setSearch}
+      topic={topic}
+      setTopic={setTopic}
+      parsedTutorials={parsedTutorials}
+    />
+  ) : (
+    <DesktopTutorialList
+      search={search}
+      setSearch={setSearch}
+      topic={topic}
+      setTopic={setTopic}
+      parsedTutorials={parsedTutorials}
+    />
+  )
+}
+
+const MobileTutorialList: FC<{
+  search: string
+  setSearch: React.Dispatch<React.SetStateAction<string>>
+  topic: TopicDropdown
+  setTopic: React.Dispatch<React.SetStateAction<TopicDropdown>>
+  parsedTutorials: Tutorial[]
+}> = ({ search, setSearch, topic, setTopic, parsedTutorials }) => {
+  return (
+    <Box sx={{ mt: 1 }}>
+      <Grid container direction="column" rowSpacing={2}>
+        <SearchBar setSearch={setSearch} />
+        <TopicFilter topic={topic} setTopic={setTopic} />
+
+        {parsedTutorials
```

Comment:
This is probably fine for now but when the tutorial list size increases you might want to debounce the input on the search box to avoid a lot of re-renders.